### PR TITLE
use go-git for cloning

### PR DESCRIPTION
**Pull request by anant2614**

### CODE REVIEW

**Comment by anant2614** [comment]
This refers to #18 . @zricethezav can you please review?

**Comment by zricethezav** [comment]
This is great, thanks! I'll look at it more closely this evening. 👍

File: leaks.go [hunk 1]
```diff
@@ -25,14 +26,25 @@ func start(opts *Options) {
...
+	}
+
+	repoName := getLocalRepoName(opts.RepoURL)
 	fmt.Printf("Cloning \x1b[37;1m%s\x1b[0m...\n", opts.RepoURL)
-	err := exec.Command("git", "clone", opts.RepoURL).Run()
+	r,err := git.PlainClone(dir + "/" + repoName, false, &git.CloneOptions{
+		URL:               opts.RepoURL,
+		RecurseSubmodules: git.DefaultSubmoduleRecursionDepth,
+	})
+	log.Printf("Cloned the repo::",r)
```
Comment (zricethezav) [inline leaks.go]: lets remove this log as the "Evaluating" message indicates a successfully cloned repo

File: leaks.go [hunk 2]
```diff
@@ -25,14 +26,25 @@ func start(opts *Options) {
...
 	if err != nil {
 		log.Printf("failed to clone repo %v", err)
```
Comment (zricethezav) [inline leaks.go]: before the "failed to clone repo" message, call `cleanup` to remove the directory. I was doing some testing and noticed even if go-git fails to clone the directory will still be made

**Comment by anant2614** [comment]
@zricethezav the PR has been updated with the requested changes.

File: leaks.go [hunk 3]
```diff
@@ -25,14 +26,25 @@ func start(opts *Options) {
 	c := make(chan os.Signal, 2)
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
 
+	//get current working directory
+	dir,err := os.Getwd()
```
Comment (zricethezav) [inline leaks.go]: one more change, use appRoot instead of dir

File: leaks.go [hunk 4]
```diff
@@ -27,15 +27,15 @@ func start(opts *Options) {
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
 
 	//get current working directory
-	dir,err := os.Getwd()
+	appRoot, err := os.Getwd()
```
Comment (zricethezav) [inline leaks.go]: appRoot is already defined here: https://github.com/zricethezav/gitleaks/blob/master/main.go#L24

Comment (anant2614) [reply leaks.go]: Ya,changing again.

**Pull request merged**